lib/data/services/registration_service_impl.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:water_tracker/data/services/registration_service.dart';
+
+class RegistrationServiceFirebaseImpl extends RegistrationService {
+  final credential = FirebaseAuth.instance;
+
+  @override
+  Future<void> registerUser(String password, String email) async {
+    credential.createUserWithEmailAndPassword(
```

Comment:
you are not awaiting it

Reply:
also, createUserWithEmailAndPassword method returns some model. Check it out in order to find out is it contains any useful information for you. For example, it is contains token.

Reply:
Yes, it contains a token and user id